src/misc_functions/returnhelper.ts

```diff
@@ -38,6 +38,18 @@ export function isSuccessful<T, E extends BCE = CE>(
     return input.kind === success;
 }
 
+export function returnSwitch<T, E extends BCE = CE, K = undefined>(
```

Comment:
I'm not sure that this function is needed. In all the locations you use it, you run `helper.merge(v)` on the result anyway. It would probably be fewer characters just to use an if-else, and that would take up fewer lines as it minimises close brackets.